### PR TITLE
19 add the ability to set the recurrence of cycle in month and years not just days

### PR DESCRIPTION
**Pull request by reichie020212**

### CODE REVIEW

**Review by jeremi — CHANGES_REQUESTED** [review]
Where are the tests?

File: g2p_programs/__manifest__.py [hunk 1]
```diff
@@ -63,6 +63,12 @@
     },
     "demo": [],
     "images": [],
+    "external_dependencies": {
+        "python": [
+            "python-dateutil",
+            "freezegun",
```
Comment (jeremi) [inline g2p_programs/__manifest__.py]: Freezegun is just needed for running the tests, so it should not be needed here.

File: g2p_programs/tests/test_cycle_manager_new_cycle.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            "month_by": "day",
+            "byday": "1",
+            "weekday": "MON",
+        }
+
+        program_manager = self.create_program_manager(**program_manager_data)
+
+        first_cycle = program_manager.new_cycle()
+
+        self.assertEqual(first_cycle.start_date, datetime.now().date())
```
Comment (jeremi) [inline g2p_programs/tests/test_cycle_manager_new_cycle.py]: This should not be the case as `datetime.now()` is not the first Monday of the month, no?
Also do not use `datetime.now()` but hardcode the dates or use constants

File: g2p_programs/tests/test_cycle_manager_new_cycle.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            "rrule_type": "weekly",
+            "cycle_duration": 1,
+            "mon": True,
+        }
+
+        program_manager = self.create_program_manager(**program_manager_data)
+
+        first_cycle = program_manager.new_cycle()
+
+        self.assertEqual(first_cycle.start_date, datetime.now().date())
```
Comment (jeremi) [inline g2p_programs/tests/test_cycle_manager_new_cycle.py]: Dec 15th is not a Monday

**Review by jeremi — CHANGES_REQUESTED** [review]

**Comment by jeremi** [comment]
@reichie020212 
- Rename `Cycle Duration` to `Recurrence`
- If I leave the default settings (30 days) why did the cycle start yesterday? That is a bug.
- By default set it monthly on the 1st
- Is there is missing recurrence config for yearly?

**Comment by reichie020212** [comment]
> @reichie020212
> 
> * Rename `Cycle Duration` to `Recurrence`
> * If I leave the default settings (30 days) why did the cycle start yesterday? That is a bug.
> * By default set it monthly on the 1st
> * Is there is missing recurrence config for yearly?

@jeremi finished the following:
- renaming
- modify default values
- fixed bug

Upon checking, there is no missing recurrence config for yearly.

**Review by jeremi — APPROVED** [review]

**Review by gonzalesedwin1123 — APPROVED** [review]

**Review by shibu-narayanan — APPROVED** [review]

**Pull request merged**